### DISCUSSION TITLE
Installing Debug Symbols

### DISCUSSION THREAD

**Discussion by toktarev** [async-profiler/async-profiler#500] [Q&A] [answered]
Hello.

Reading https://github.com/jvm-profiling-tools/async-profiler#installing-debug-symbols this part of the documentation I have a question.

Does it affect performance if I use openjdk-8-dbg instead if openjdk-8 ?

Is it safe to switch production on openjdk-8-dbg ?

Thanks.

**Comment by apangin** [comment, marked as answer]
`openjdk-8-dbg` package contains *additional* files with symbols. It does not change Java binaries used to run the program.

Make sure not to confuse debuginfo with debug JVM though.

**Comment by toktarev** [comment]
Thanks !

**Comment by toktarev** [comment]
Just for the clarification

- with debug JVM though

Do you mean -Xdebug option here? Or native build with -NDebug option ?

I was worried that "openjdk-8-dbg" is a special build with enabled internal addition assertion for example,
or without additional c++  native optimizations (e.g. -O3 ).

It can potentially affect performance.

**Reply by apangin** [reply]
I meant, with debug builds of the VM (compiled with `--enable-debug`).
Never mind, `openjdk-8-dbg` contains only the separate files with debug symbols, you can [check it yourself](https://packages.ubuntu.com/focal/amd64/openjdk-8-dbg/filelist).

**Comment by toktarev** [comment]
Thanks.

Actually, thought the same (it would be strange if you'd recommend VM (compiled with --enable-debug)
for production).

Reinsured.
